st2common/st2common/operators.py

```diff
@@ -161,6 +161,103 @@ def search(value, criteria_pattern, criteria_condition, check_function):
     return rtn
 
 
+def multiple(value, criteria_pattern, criteria_condition, check_function):
+    """
+    Allow comparison of payload items to multiple criteria using different logicial conditions.
+    Performs same function as the "search" operator and contains additional features.
+
+    value: the payload items
+    condition: one of:
+      * all2all - true if all payload items match all criteria items
+      * all2any - true if all payload items match any criteria items
+      * any2any - true if any payload items match any criteria items
+      * any2all - true if any payload items match all criteria items
+      * all - same as all2all (useful to maintain backward compatibility with search operator)
+      * any - same as any2all (useful to maintain backward compatibility with search operator)
```

Comment:
I could go either way. I picked multiple because it was slightly more descriptive of what the feature does. If the team likes it that way, I imagine search will eventually be deprecated.

Reply:
I'd prefer to rely on the existing interface and naming for backward compatibility and stable interface.
Please move the new functionality under the `search` operator.